File: Excel and OleDb basics to advance operations/[C#][VB]-Excel and OleDb basics to advance operations/C# and VB/OleDbHelper/Utility.vb
Imports System.Data.OleDb

Public Class Utility
    Public Sub New()
    End Sub
    Public Function TableExists(ByVal ConnectionString As String, ByVal TableName As String) As Boolean
        Dim Result As Boolean = False

        Using cn As New OleDbConnection(ConnectionString)
            cn.Open()
            Dim dt As DataTable = New DataTable With {.TableName = "test"}
            dt = cn.GetOleDbSchemaTable(OleDbSchemaGuid.Tables, New Object() {Nothing, Nothing, Nothing, "TABLE"})
            cn.Close()

            Dim query = (From F In dt.Rows.Cast(Of DataRow)() Where F.Field(Of String)("TABLE_NAME").ToString = TableName).FirstOrDefault

            If query IsNot Nothing Then
                Result = True
            End If

        End Using

        Return Result

    End Function
    ''' <summary>
    ''' Returns a list of sheet names from Excel or table names from Access
    ''' </summary>
    ''' <param name="ConnectionString"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Function SheetNames(ByVal ConnectionString As String) As List(Of SheetNameData)
        Dim Names As New List(Of SheetNameData)

        Using cn As New OleDbConnection(ConnectionString)
            cn.Open()
            Dim dt As DataTable = New DataTable With {.TableName = "AvailableSheetsTables"}
            dt = cn.GetOleDbSchemaTable(OleDbSchemaGuid.Tables, New Object() {Nothing, Nothing, Nothing, "TABLE"})
            cn.Close()

            Names =
                (
                    From F In dt.Rows.Cast(Of DataRow)()
                    Select New SheetNameData With
                           {
                               .DisplayName = F.Field(Of String)("TABLE_NAME").Replace("$", ""),
                               .ActualName = F.Field(Of String)("TABLE_NAME")
                           }
                ).ToList

        End Using

        Return Names
    End Function
    ''' <summary>
    ''' Get column names for a connection that has HDR = Yes
    ''' </summary>
    ''' <param name="ConnectionString"></param>
    ''' <param name="SheetName"></param>
    ''' <returns></returns>
    ''' <remarks>
    ''' You could use this with HDR = No but here we are
    ''' using it with HDR = Yes
    ''' 
    ''' Also, we can get column schema from the connection and 
    ''' order column names by Ordinal_Position so that Column_Name
    ''' column will not be ordered.
    ''' </remarks>
    Public Function GetColumnNames(ByVal ConnectionString As String, ByVal SheetName As String) As List(Of String)
        Dim ColumnNames As New List(Of String)
        Using cn As New OleDbConnection(ConnectionString)
            cn.Open()
            Using cmd As New OleDbCommand(String.Format("select * from [{0}]", SheetName), cn)
                Using reader = cmd.ExecuteReader(CommandBehavior.SchemaOnly)
                    Dim dt As DataTable = reader.GetSchemaTable()
                    Dim nameCol = dt.Columns("ColumnName")
                    For Each row As DataRow In dt.Rows
                        ColumnNames.Add(row(nameCol).ToString)
                    Next
                End Using
            End Using
        End Using
        Return ColumnNames
    End Function
End Class
Public Class SheetNameData
    Public Sub New()

    End Sub
    Public Property DisplayName As String
    Public Property ActualName As String

End Class

File: Export MS-Access table to MS-Excel Worksheet/[C#][VB]-Export MS-Access table to MS-Excel Worksheet/C# and VB/ExcelOleDbLibrary/Utility.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Data.OleDb;
using System.Data;

namespace ExcelOleDbLibrary
{
    /// <summary>
    /// Taken from another MSDN code sample I wrote.
    /// </summary>
    public class Utility
    {
        public Utility()
        {
        }
        public bool TableExists(string ConnectionString, string TableName)
        {
            bool Result = false;

            using (OleDbConnection cn = new OleDbConnection(ConnectionString))
            {
                cn.Open();
                DataTable dt = new DataTable { TableName = "test" };
                dt = cn.GetOleDbSchemaTable(OleDbSchemaGuid.Tables, new object[] { null, null, null, "TABLE" });
                cn.Close();

                var query = (
                    from F in dt.Rows.Cast<DataRow>()
                    where F.Field<string>("TABLE_NAME").ToString() == TableName
                    select F).FirstOrDefault();

                if (query != null)
                {
                    Result = true;
                }

            }

            return Result;

        }
        /// <summary>
        /// Returns a list of sheet names from Excel or table names from Access
        /// </summary>
        /// <param name="ConnectionString"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public List<SheetNameData> SheetNames(string ConnectionString)
        {
            List<SheetNameData> Names = new List<SheetNameData>();

            using (OleDbConnection cn = new OleDbConnection(ConnectionString))
            {

                cn.Open();

                DataTable dt = new DataTable { TableName = "AvailableSheetsTables" };
                dt = cn.GetOleDbSchemaTable(OleDbSchemaGuid.Tables, new object[] { null, null, null, "TABLE" });


                Names = 
                    (
                        from F in dt.Rows.Cast<DataRow>()
                        select new SheetNameData
                            {
                                DisplayName = F.Field<string>("TABLE_NAME").Replace("$", ""),
                                ActualName = F.Field<string>("TABLE_NAME")
                            }
                    ).ToList();

            }

            return Names;

        }
        /// <summary>
        /// Get column names for a connection that has HDR = Yes
        /// </summary>
        /// <param name="ConnectionString"></param>
        /// <param name="SheetName"></param>
        /// <returns></returns>
        /// <remarks>
        /// You could use this with HDR = No but here we are
        /// using it with HDR = Yes
        /// 
        /// Also, we can get column schema from the connection and 
        /// order column names by Ordinal_Position so that Column_Name
        /// column will not be ordered.
        /// </remarks>
        public List<string> GetColumnNames(string ConnectionString, string SheetName)
        {
            List<string> ColumnNames = new List<string>();
            using (OleDbConnection cn = new OleDbConnection(ConnectionString))
            {
                cn.Open();
                using (OleDbCommand cmd = new OleDbCommand(string.Format("select * from [{0}]", SheetName), cn))
                {
                    using (var reader = cmd.ExecuteReader(CommandBehavior.SchemaOnly))
                    {
                        DataTable dt = reader.GetSchemaTable();
                        var nameCol = dt.Columns["ColumnName"];
                        foreach (DataRow row in dt.Rows)
                        {
                            ColumnNames.Add(row[nameCol].ToString());
                        }
                    }
                }
            }

            return ColumnNames;

        }
    }
}
